Description: js/ui/background.js: Match 19.10 Yaru purple login screen with noise.
 This way the login animation will appear to expand over the login
 screen (system background) instead of suddenly replacing it.
Original author: Didier Roche <didrocks@ubuntu.com>
Author: Daniel van Vugt <daniel.van.vugt@canonical.com>
Last-Update: 2019-08-30
Forwarded: not-needed

Index: gnome-shell/js/ui/background.js
===================================================================
--- gnome-shell.orig/js/ui/background.js
+++ gnome-shell/js/ui/background.js
@@ -96,6 +96,7 @@
 const { Clutter, GDesktopEnums, Gio, GLib, GnomeDesktop, Meta } = imports.gi;
 const Signals = imports.signals;
 
+const Desktop = imports.misc.desktop;
 const LoginManager = imports.misc.loginManager;
 const Main = imports.ui.main;
 const Params = imports.misc.params;
@@ -501,6 +502,27 @@ var SystemBackground = class SystemBackg
     constructor() {
         let file = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/noise-texture.png');
 
+        // Yaru is default ubuntu theme, match GDM look
+        if (Desktop.is("ubuntu")) {
+            if (_systemBackground == null) {
+                _systemBackground = new Meta.Background({ meta_display: global.display })
+                // Yaru uses darken(#762572, 10%) + noise for #lockDialogGroup
+                let [, ubuntuColor] = Clutter.Color.from_string('#4f194c');
+                _systemBackground.set_color(ubuntuColor);
+                _systemBackground.set_file(file, GDesktopEnums.BackgroundStyle.WALLPAPER);
+            }
+            this.actor = new Meta.BackgroundActor({ meta_display: global.display,
+                                                    monitor: 0,
+                                                    background: _systemBackground });
+            let id = GLib.idle_add(GLib.PRIORITY_DEFAULT, () => {
+                this.emit('loaded');
+                return GLib.SOURCE_REMOVE;
+            });
+            GLib.Source.set_name_by_id(id, '[gnome-shell] SystemBackground.loaded');
+
+            return;
+        }
+
         if (_systemBackground == null) {
             _systemBackground = new Meta.Background({ meta_display: global.display });
             _systemBackground.set_color(DEFAULT_BACKGROUND_COLOR);
